### PR TITLE
fix: Federated identity subject

### PR DESCRIPTION
**Pull request by jacopocarlini**
### List of changes

🐛  if you choose `pull_request` as subject the right pattern is `repo:OWNER/REPO_NAME:pull_request`
(👀 see this [run](https://github.com/pagopa/pagopa-print-payment-notice-service/actions/runs/9158612132/job/25177329197?pr=14#:~:text=Error%3A%20%3A%20AADSTS700213,19) for the error)

### Motivation and context

There are 3 options with the subject:

#### Ref
`repo:pagopa/pagopa-print-payment-notice-service:ref:refs/tags/1.0.0
`

#### Environment
`repo:pagopa/pagopa-print-payment-notice-service:environment:dev
`

#### Pull Request
`repo:pagopa/pagopa-print-payment-notice-service:pull_request
`

### Type of changes

- [ ] Add new module
- [x] Update existing module
- [ ] Remove existing module

### Does this introduce a breaking change?

- [ ] Yes
- [x] No

### Run checks

Useful commands to run checks on local machine

```sh
bash .utils/terraform_run_all.sh init local
pre-commit run -a
```

### CODE REVIEW

File: github_federated_identity/main.tf [hunk 1]
```diff
@@ -48,5 +48,5 @@ resource "azurerm_federated_identity_credential" "identity_credentials" {
   audience            = each.value.audience
   issuer              = each.value.issuer
   parent_id           = azurerm_user_assigned_identity.identity.id
-  subject             = "repo:${each.value.org}/${each.value.repository}:${each.value.credentials_scope}:${each.value.subject}"
+  subject             = each.value.subject == "environment" ? "repo:${each.value.org}/${each.value.repository}:${each.value.credentials_scope}:${each.value.subject}" : "repo:${each.value.org}/${each.value.repository}:${each.value.subject}"
```
Comment (Krusty93) [inline github_federated_identity/main.tf:51]: Is it possible to create a federation without a subject? As far as I know, an environment/branch/tag must be supplied

Comment (gunzip) [inline github_federated_identity/main.tf:51]: can you add a comment on why is this needed?

Comment (jacopocarlini) [reply github_federated_identity/main.tf:51]: I update the PR description with the info

**Comment by Krusty93** [comment]
Just a matter of curiosity, in which scenario do you need a user managed identity for a pull request only?

**Comment by jacopocarlini** [comment]
I want to run integration/smoke tests when a PR is open, but if I use the GH environment a deployment is created automatically (bug ref https://github.com/actions/runner/issues/2120). So I took the `client_id` from the repo secrets

**Comment by Krusty93** [comment]
> @Krusty93 I want to run integration/smoke tests when a PR is open, but if I use the GH environment a deployment is created automatically (bug ref [actions/runner#2120](https://github.com/actions/runner/issues/2120)). So I took the `client_id` from the repo secrets

Is there any problem creating a deployment at each workflow run as far as you know?

**Comment by jacopocarlini** [comment]
> > @Krusty93 I want to run integration/smoke tests when a PR is open, but if I use the GH environment a deployment is created automatically (bug ref [actions/runner#2120](https://github.com/actions/runner/issues/2120)). So I took the `client_id` from the repo secrets
> 
> Is there any problem creating a deployment at each workflow run as far as you know?

Because Jira marks the task as deployed in DEV/UAT/PROD when we run integration test on a branch.
I want to use this subject as a workaround.

image


In any case this PR fixes a bug if someone wants to use this kind of subject

**Comment by Krusty93** [comment]
> > > @Krusty93 I want to run integration/smoke tests when a PR is open, but if I use the GH environment a deployment is created automatically (bug ref [actions/runner#2120](https://github.com/actions/runner/issues/2120)). So I took the `client_id` from the repo secrets
> 
> > 
> 
> > Is there any problem creating a deployment at each workflow run as far as you know?
> 
> 
> 
> Because Jira marks the task as deployed in DEV/UAT/PROD when we run integration test on a branch.
> 
> I want to use this subject as a workaround.
> 
> 
> 
> image
> 
> 
> 
> 
> 
> In any case this PR fixes a bug if someone wants to use this kind of subject
> 
> 

Yup, PR is fine to me since your first reply, I asked trying to understand your needs and eventually implementing the same solution in other repos

**Review by mamari90 — APPROVED** [review]
